### PR TITLE
kelly/lookback_tick_trade

### PR DESCRIPTION
**Pull request by kellybinary**

### CODE REVIEW

File: src/javascript/app/pages/trade/purchase.js [hunk 1]
```diff
@@ -164,6 +164,8 @@ const Purchase = (() => {
             let category = sessionStorage.getItem('formname');
             if (/^(risefall|higherlower)$/.test(category)) {
                 category = 'callput';
+            } else if (/^lookback(highlow|high|low)$/.test(category)) {
```
Comment (negarn) [inline src/javascript/app/pages/trade/purchase.js]: why not just `/^lookback/`?

File: src/javascript/app/pages/trade/tick_trade.js [hunk 2]
```diff
@@ -272,10 +274,13 @@ const TickDisplay = (() => {
                 const x            = status === 'open' ? 60 : (chart.chartWidth - width - 45);
                 const is_resetcall = contract.contract_type === 'RESETCALL';
                 const has_reset    = current_barrier !== entry_barrier;
+
+                drawBarrier({ id: 'tick-barrier', value: entry_barrier, text: 'Barrier', label: { align: 'right',  x: -x }, dashStyle: has_reset ? 'dot' : 'default', textBottom: has_reset && !is_resetcall });
+                contract_barrier_label = getContractBarrierLabel('Barrier', entry_barrier);
```
Comment (negarn) [inline src/javascript/app/pages/trade/tick_trade.js]: why not put this in else?

File: src/javascript/app/pages/trade/tick_trade.js [hunk 3]
```diff
@@ -244,7 +243,7 @@ const TickDisplay = (() => {
             // round calculated barrier
             const calc_barrier = (total / applicable_ticks.length).toFixed(parseInt(display_decimals) + 1);
             drawBarrier({ value: calc_barrier, text: 'Average' });
-            contract_barrier_label = getContractBarrierLabel('Average', calc_barrier);
+            contract_barrier_label = getContractBarrierLabel('Barrier', calc_barrier);
```
Comment (negarn) [inline src/javascript/app/pages/trade/tick_trade.js]: we're changing the label to barrier?

File: src/javascript/app/pages/trade/tick_trade.js [hunk 4]
```diff
@@ -275,29 +254,82 @@ const TickDisplay = (() => {
...
-                        dashStyle: 'dash',
-                    });
+            if (entry_barrier && current_barrier) {
+                const width        = ($('.highcharts-markers')[0].getBoundingClientRect() || {}).width;
+                const x            = status === 'open' ? 60 : (chart.chartWidth - width - 45);
+                const is_resetcall = contract.contract_type === 'RESETCALL';
+                const has_reset    = current_barrier !== entry_barrier;
+
+                drawBarrier({ id: 'tick-barrier', value: entry_barrier, text: 'Barrier', label: { align: 'right',  x: -x }, dashStyle: has_reset ? 'dot' : 'default', textBottom: has_reset && !is_resetcall });
+                contract_barrier_label = getContractBarrierLabel('Barrier', entry_barrier);
```
Comment (negarn) [inline src/javascript/app/pages/trade/tick_trade.js]: why not put this in else?

File: src/javascript/app/pages/trade/tick_trade.js (hunk 3 above)
Comment (kellybinary) [reply src/javascript/app/pages/trade/tick_trade.js]: for Asians trade types,  it is currently labelled as Barrier in the overlay. but i can change it to Average instead

Comment (negarn) [reply src/javascript/app/pages/trade/tick_trade.js]: let's keep it :)

File: src/javascript/app/pages/trade/lookback.js [hunk 5]
```diff
@@ -47,11 +47,27 @@ const Lookback = (() => {
...
+    // High-Low  : high_barrier - low_barrier
+    const getContractPayout = ({ contract_type, multiplier, barrier, current_spot, high_barrier, low_barrier }) => { 
+        const barrier_map = {
+            LBFLOATCALL: [current_spot,  barrier],
+            LBFLOATPUT : [barrier,       current_spot],
+            LBHIGHLOW  : [high_barrier,  low_barrier],
+        };
+        const [barrier1, barrier2] = barrier_map[contract_type];
+        const payout = +multiplier * (+barrier1 - +barrier2);
+        return payout < 0 ? 0 : payout;
```
[the remark this answers is not in the document]
Comment (negarn) [reply src/javascript/app/pages/trade/lookback.js]: oh yeah, math.max

**Pull request closed without merging**